File: content/learn/arrays-easy/04-remove-duplicates/solution.js
function removeDuplicates(a) {
  if (a.length === 0) return 0;
  let write = 0;
  for (let read = 1; read < a.length; read++) {
    if (a[read] !== a[write]) {
      write++;
      a[write] = a[read];
    }
  }
  return write + 1;
}

const n = parseInt(readline(), 10);
const a = n > 0 ? readline().split(" ").map(Number) : [];
const k = removeDuplicates(a);
print(k);
print(a.slice(0, k).join(" "));

File: content/learn/arrays-easy/04-remove-duplicates/solution.py
def remove_duplicates(a):
    if not a:
        return 0
    write = 0
    for read in range(1, len(a)):
        if a[read] != a[write]:
            write += 1
            a[write] = a[read]
    return write + 1


n = int(input())
a = list(map(int, input().split())) if n > 0 else []
k = remove_duplicates(a)
print(k)
print(" ".join(str(x) for x in a[:k]))
